### PR TITLE
feat(sma): implement re-design v2 SMA API

### LINKED ISSUE #3524: [v2 sys-mgmt-agent] v2 API redesign
# 🚀 Feature Request

### Relevant Package

 This feature request is for sys-mgmt-agent

### Description
followed by https://github.com/edgexfoundry/edgex-go/issues/3418#issuecomment-849724441
the new design includes:
- use comma-separated query parameter 
- 207 responses

this will affect `api/v2/health`, `api/v2/metrics` and `api/v2/configs`

### PR DESCRIPTION
**Pull request by hahattan**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-go/blob/master/.github/Contributing.md.

## What is the current behavior?

1. requested services is specified in the URL: `/api/v2/health/core-data,core-metadata`
2. returned status (200/404) could not represent the actual response of each individual requested service
```
curl http://localhost:58890/api/v2/health/core-data,core-metadata,invalid
200
{
    "apiVersion": "v2",
    "statusCode": 200,
    "health": {
        "core-data": "healthy",
        "core-metadata": "healthy",
        "invalid": "invalid service is not registered. Might not have started... "
    }
}
```

## What is the new behavior?
1. use query parameter instead: `/api/v2/system/health?services=core-data,core-metadata`
2. return 207
```
curl http://localhost:58890/api/v2/system/health?services=core-data,core-metadata,invalid
207
[
    {
        "apiVersion": "v2",
        "message": "invalid service is not registered. Might not have started... ",
        "statusCode": 404,
        "serviceName": "invalid"
    },
    {
        "apiVersion": "v2",
        "statusCode": 200,
        "serviceName": "core-data"
    },
    {
        "apiVersion": "v2",
        "statusCode": 200,
        "serviceName": "core-metadata"
    }
]
```

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## New Imports

- [ ] Yes
- [x] No

## Specific Instructions
Are there any specific instructions or things that should be known prior to reviewing?

## Other information

### CODE REVIEW

File: internal/system/agent/v2/application/direct/health.go [hunk 1]
```diff
@@ -6,33 +6,54 @@
...
+func GetHealth(services []string, rc registry.Client) []interface{} {
+	var mu sync.Mutex
+	var wg sync.WaitGroup
+	var responses []interface{}
 
-func GetHealth(services []string, registryClient registry.Client) map[string]string {
-	health := make(map[string]string)
 	for _, service := range services {
-		if registryClient == nil {
-			health[service] = "registry is required to obtain service health status."
```
Comment (cloudxxx8) [inline internal/system/agent/v2/application/direct/health.go:20]: We need to keep this checking logic to check whether registry client is available for this function, and it should be moved to the beginning of this function.

**Review by cloudxxx8 — CHANGES_REQUESTED** [review]

File: internal/system/agent/v2/application/direct/health.go (hunk 1 above)
Comment (hahattan) [reply internal/system/agent/v2/application/direct/health.go:20]: the logic is moved to controller layer

Comment (cloudxxx8) [reply internal/system/agent/v2/application/direct/health.go:20]: this is not protocol specific logic, so it should be located in the application layer.
we might implement other controller protocol in the future, and this logic should be common.

**Review by cloudxxx8 — APPROVED** [review]
LGTM

File: internal/system/agent/v2/controller/http/controller.go [hunk 2]
```diff
@@ -37,48 +35,65 @@ func NewAgentController(dic *di.Container) *AgentController {
...
-	services := strings.Split(vars[v2.Services], v2.CommaSeparator)
 
+	queryString := r.URL.Query()[v2.Services]
+	if queryString == nil || (len(queryString) == 1 && queryString[0] == "") {
+		err := errors.NewCommonEdgeX(errors.KindContractInvalid, "failed to parse query", nil)
+		utils.WriteErrorResponse(w, r.Context(), lc, err, "")
+		return
+	}
+
+	services := strings.Split(queryString[0], v2.CommaSeparator)
```
Comment (lenny-intel) [inline internal/system/agent/v2/controller/http/controller.go:70]: Make a helper function to do this to reduce duplicate code

**Review by lenny-intel — CHANGES_REQUESTED** [review]

**Review by lenny-intel — APPROVED** [review]
LGTM

**Pull request merged**